### ISSUE TITLE
"possibly unbound" warning when redefining builtins

### ISSUE THREAD

**Issue by ces42** [microsoft/pyright#4074]
**Describe the bug**
The following code produces a error/warning:
```python
if input():
    print = lambda *x: None

print('a')  # "print" is possibly unbound
```
If I add a line `print = print` at the top the warning goes away

**Expected behavior**
Builtin names should be treated as if they were defined at the top of the file so this error shouldn't appear.

**VS Code extension or command-line**
I'm running `pyright 1.1.273` inside neovim

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.277, which I just published. It will also be included in a future release of pylance.
